notebooks/ch16/classification_precision_recall.md

```diff
@@ -0,0 +1,326 @@
+        plt.text(j, i, format(cm[i, j], fmt),
+                 horizontalalignment="center",
+                 color="white" if cm[i, j] > thresh else "black")
+
+    plt.tight_layout()
+    plt.ylabel('True label')
+    plt.xlabel('Predicted label')
+    
+class_names = ['Ham', 'Spam']
+# Plot non-normalized confusion matrix
+plt.figure()
+plot_confusion_matrix(cnf_matrix, classes=class_names,
+                      title='Confusion matrix, without normalization')
+
+# # Plot normalized confusion matrix
+# plt.figure()
+# plot_confusion_matrix(cnf_matrix, classes=class_names, normalize=True,
+#                       title='Normalized confusion matrix')
+
+```
+
+    Confusion matrix, without normalization
+    [[710   4]
+     [ 33   3]]
+
+
+
+![png](classification_precision_recall_files/classification_precision_recall_11_1.png)
+
+
+See [here](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html#sklearn.metrics.confusion_matrix) for more on scikit-learn's confusion matrix generator.
+
+Adding together the number of true positives and false negatives, we know the training dataset contains 36 spam emails, but the model only detected 3 of them. Despite its high accuracy, the model identified $\frac{1}{12}$ of spam emails, which is quite a poor performance. 
+
+What if we used a classifier `ham_only` that simply labelled every email as ham? We can calculate its accuracy simply by knowing the number of ham emails in the dataset: $$ \text{Accuracy} = \frac{TP + TN}{TP + FP + TN + FN} = \frac{0 + 714}{0 + 0 + 714 + 36} = .95$$
+
+
+This model, which is essentially useless as a classifier, surprisingly achieves a better accuracy than our EDA-driven model. Clearly accuracy alone can be a misleading indicator of model performance. This is especially characteristic of **class-imbalanced datasets**, in which a vast majority of labels belong in one class over the other. In this case, most of our emails are ham. The confusion matrix shows that the model based on the occurrences of three words had many more false negatives than true positives, which implies that it didn't have enough features to reliably identify spam, but because the dataset is predominantly ham, it had a deceptively high accuracy. Another common example of class imbalance is disease detection when the frequency of the disease in a population is low.
+
+We now turn to precision and recall, which are better suited metrics for evaluating class-imbalanced datasets.
+
+### Precision
+
+**Precision** (also called **positive predictive value**) measures the proportion of positively classified data that are actually positive. 
+
+$$ \text{Precision} = \frac{TP}{TP + FP} $$ 
+
+A model with high precision accurately classifies data that truly has positive labels, whereas a model with low precision incorrectly picks out aspects of the positive class in truly negative data with high frequency (many false positives).
```

Comment:
"incorrectly picks out aspects of the positive class in truly negative data" was a little confusing for me. It's theoretically sound - since it incorrectly picks up trends that it thinks is "positive" in truly negative data - but I think for readers, it's better to just define what the classifier's actually doing:

> whereas a model with low precision incorrectly labels truly negative data as positive (i.e. false positive).